grasa_event_locator/views.py

```diff
@@ -483,27 +481,56 @@ def resetpw(request):
                 if User.objects.filter(username=request.POST['emailAddr']).exists():
                         resetPWURLs.objects.get(user_ID = request.POST['emailAddr']).delete()
                         resetlink = ''.join([random.choice(string.ascii_letters + string.digits) for n in range(15)])
-                        resetPWURL = resetPWURLs(user_ID = request.POST['emailAddr'], reset_string= resetlink)
+                        resetPWURL = resetPWURLs(user_ID = request.POST['emailAddr'], reset_string= resetlink, expiry_time= (datetime.now() + timedelta(minutes = 60)))
                         resetPWURL.save()
                         # Make sure to pull the hostname from config file.
-                        print(send_email([request.POST['emailAddr']], "GRASA - Reset Password", "You've requested a password reset at the GRASA Event Locator. Please visit this linnk: http://grasa.larrimore.de/resetPWForm/" + resetlink))
+                        print(send_email([request.POST['emailAddr']], "GRASA - Reset Password", "You've requested a password reset at the GRASA Event Locator. Please visit this link: http://grasa.larrimore.de/resetPWForm/" + resetlink + ". This link expires in an hour."))
                 context = {'email_submitted': True}
                 return render(request, 'resetPW.html', context)
         else:
             return render(request, 'resetPW.html')
 
 def resetPWForm(request, reset_string):
-        if request.method == 'POST' and request.POST['new'] == request.POST['confirm']:
-                print(reset_string)
-                username = resetPWURLs.objects.get(reset_string=reset_string)
-                print(username)
-                user = User.objects.get(username=username.user_ID)
-                print(user)
-                user.set_password(request.POST['new'])
-                user.save()
-                resetPWURLs.objects.get(user_ID = username.user_ID).delete()
-                return redirect("login_page")
+        try:
+            #Check if the current time is greater than the timestamp in the table (which is 60 minutes after submission)
+            if datetime.now() > datetime.strptime(resetPWURLs.objects.get(reset_string=reset_string).expiry_time, '%Y-%m-%d %H:%M:%S.%f'):
+                #If so, show the expired message, hide the form.
+                context = {'expired': True, "valid_string" : False}
+                # Then delete the associated table entry that's out of date.
+                with connection.cursor() as cursor:
+                    cursor.execute(
+                        "DELETE FROM `grasa_event_locator_resetpwurls` WHERE `user_ID` = '" + resetPWURLs.objects.get(
+                            reset_string=reset_string).user_ID + "';")
```

Comment:
Hmm. Is there a safer way to write this than with the raw SQL? cc: @eguy006

Reply:
Good point. Just did a new commit, removing the SQL and adding in the correct python commands.